src/ansys/fluent/core/session_solver.py

```diff
@@ -228,3 +230,47 @@ def read_case(self, file_name: str, lightweight_mode: bool = False):
             fut.add_done_callback(functools.partial(Solver._sync_from_future, self))
         else:
             self.file.read(file_type="case", file_name=file_name)
+
+    def read_case(
+        self,
+        file_name: str,
+    ):
+        """Reads a case file.
+
+        Parameters
+        ----------
+        file_name : str
+            Case file name
+        """
+        if pypim.is_configured():
+            self._pypim_upload(
+                file_name=file_name,
+                on_uploaded=(
+                    lambda file_names: self.file.read_case(file_name=file_names)
+                ),
+            )
+        else:
+            self._no_pypim_helper(
+                file_name,
+                api=(lambda file_names: self.file.read_case(file_name=file_names)),
+            )
+
+    def write_case(
+        self,
+        file_name: str,
+    ):
+        """Writes a case file.
+
+        Parameters
+        ----------
+        file_name : str
+            Case file name
+        """
+        self._no_pypim_helper(
+            file_name,
+            api=(lambda file_names: self.file.write_case(file_name=file_names)),
```

Comment:
Interesting that you made this plural - does this support multiple files?

Reply:
if we use singular then it shadows `file_name` from outer scope, hence we have made it singular. Another option was to use `filename` but we are not using it now.

Reply:
Any other suggestion will be helpful for this. I also don't want to use plural.